### PR TITLE
Build Tribler by using GitHub Actions

### PR DESCRIPTION
**Pull request by drew2a**
This PR adds the `build.yml` workflow for ubuntu that runs on each PR.

Also, it performs the first step of `update_version_from_git.py` refactoring: extracting `update_version.py` and `update_metadata.py`.

When I add a build workflow for each OS, `update_version_from_git.py` could be deleted.

This change is backwards-compatible with Jenkins one-click-release, but we must add the following lines to Jenkins before the call `makedist_debian.sh`:

```bash
          git describe | python -c 'import sys; print(next(sys.stdin).lstrip("v"))' > .TriblerVersion
          git rev-parse HEAD > .TriblerCommit
```

### CODE REVIEW

**Comment by drew2a** [comment]
For the history (manual approve for upload):

```python
name: Binaries

on:
  pull_request:

jobs:
  build:
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        python-version: [ '3.8' ]
        os: [ ubuntu-22.04 ]

    steps:
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0
          ref: ${{github.event.pull_request.head.sha}}

      - name: Cache package
        id: cache
        uses: actions/cache@v3
        with:
          path: cache
          key: ubuntu-cache

      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v3
        with:
          python-version: ${{ matrix.python-version }}

      - name: Install Pip dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r ./requirements-build.txt

      - name: Install Ubuntu dependencies
        run: |
          sudo apt-get update
          sudo apt-get -y install debhelper devscripts
          # for qt:
          sudo apt-get -y install libxcb-xinerama0-dev libqt5x11extras5

      - name: Save Git info
        run: |
          git describe --tags HEAD > .TriblerVersion
          echo ${{ github.event.pull_request.head.ref }} > .TriblerBranch
          echo ${{ github.event.pull_request.head.sha }} > .TriblerCommit

      - name: Run build script
        timeout-minutes: 5
        env:
          SENTRY_URL: ${{secrets.SENTRY_URL}}
          QT_QPA_PLATFORM: offscreen
          QT_ACCESSIBILITY: 1
          QT_IM_MODULE: ibus
        run: |
          ./build/debian/makedist_debian.sh

      - name: Move binaries to cache
        run: |
          mkdir -p cache
          mv tribler.deb ./cache


  upload:
    runs-on: ubuntu-latest
    needs: [ build ]
    environment:
      name: Upload
    steps:
      - name: Cache package
        id: cache
        uses: actions/cache@v3
        with:
          key: ubuntu-cache
          path: cache

      - name: Upload Artifact
        uses: actions/upload-artifact@v3
        with:
          name: tribler.deb
          path: cache/tribler.deb
          retention-days: 1

```

File: build/debian/tribler/debian/changelog [hunk 1]
```diff
@@ -1,58 +1,3 @@
-tribler (7.0.0) unstable; urgency=medium
```
Comment (xoriole) [inline build/debian/tribler/debian/changelog:1]: This is history of older versions, I'd like to keep these in place.

**Review by xoriole — CHANGES_REQUESTED** [review]
Nice improvements 👍 

A concern is how it affects the Flatpak build and push to Flathub, that needs to be checked before merging.
https://flathub.org/apps/details/org.tribler.Tribler

**Comment by xoriole** [comment]
```
          git describe | python -c 'import sys; print(next(sys.stdin).lstrip("v"))' > .TriblerVersion
          git rev-parse HEAD > .TriblerCommit
```          
These commands are added to Ubuntu `.deb` builder on Jenkins
https://jenkins-ci.tribler.org/job/Build-Tribler_release/job/Build-Ubuntu64/

File: build/debian/tribler/debian/changelog (hunk 1 above)
Comment (drew2a) [reply build/debian/tribler/debian/changelog:1]: Agree. But this history only consists of versions until the `7.0.0` which makes `changelog` a bit suspicious.
Why other versions are missed?

I see two options here:
1. To not have versions at all (this option is implemented in the current PR)
2. To have all versions

For option number 2 we should somehow create a changelog from the scratch. I've tried a few ways of doing this (unsuccessfuly) and returned back to option number 1.

If you are able to create a full changelog, feel free to add a commit to the PR or add a comment to this conversation.

**Comment by drew2a** [comment]
@xoriole I don't see any mentions of Flatpak in our repo. 

Could you please give me a link to the Flatpak build?

**Comment by xoriole** [comment]
> @xoriole I don't see any mentions of Flatpak in our repo.
> 
> Could you please give me a link to the Flatpak build?

Flatpak build is community contributed and is present here:
https://github.com/flathub/org.tribler.Tribler

Revisiting the config file, the Flatpak package is built using Debian package from the GitHub release URL. Therefore, this PR  would not have any effect on that. So, my original concern regarding Flatpak build is invalid.

File: build/debian/tribler/debian/changelog (hunk 1 above)
Comment (drew2a) [reply build/debian/tribler/debian/changelog:1]: Agree. I'll return the history.

Comment (drew2a) [reply build/debian/tribler/debian/changelog]: Done!

**Review by xoriole — APPROVED** [review]

**Pull request merged**